mdata/aggregator.go

```diff
@@ -43,31 +43,31 @@ func NewAggregator(store Store, cachePusher cache.CachePusher, key schema.AMKey,
 		case conf.Avg:
 			if aggregator.sumMetric == nil {
 				key.Archive = schema.NewArchive(schema.Sum, span)
-				aggregator.sumMetric = NewAggMetric(store, cachePusher, key, conf.Retentions{ret}, 0, nil, dropFirstChunk)
+				aggregator.sumMetric = NewAggMetric(store, cachePusher, key, conf.Retentions{ret}, 0, 0, nil, dropFirstChunk)
```

Comment:
why would we deliberately pass interval=0 and not the correct interval (`span`) ?  even if we don't end up using the value in NewAggMetric() from these calls, this makes the code brittle and likely to break once we start using it for something else.